jsonrpc/eth_endpoint.go

```diff
@@ -111,27 +111,67 @@ func (e *Eth) SendTransaction(arg *txnArgs) (interface{}, error) {
 	return transaction.Hash.String(), nil
 }
 
-// GetTransactionByHash returns a transaction by his hash
+// GetTransactionByHash returns a transaction by its hash.
+// If the transaction is still pending -> return the txn with some fields omitted
+// If the transaction is sealed into a block -> return the whole txn with all fields
 func (e *Eth) GetTransactionByHash(hash types.Hash) (interface{}, error) {
-	blockHash, ok := e.d.store.ReadTxLookup(hash)
-	if !ok {
-		// txn not found
-		return nil, nil
-	}
-	block, ok := e.d.store.GetBlockByHash(blockHash, true)
-	if !ok {
-		// block receipts not found
-		return nil, nil
+	// findSealedTx is a helper method for checking the world state
+	// for the transaction with the provided hash
+	findSealedTx := func() (*transaction, bool) {
+		// Check the chain state for the transaction
+		blockHash, ok := e.d.store.ReadTxLookup(hash)
+		if !ok {
+			// Block not found in storage
+			return nil, false
+		}
+		block, ok := e.d.store.GetBlockByHash(blockHash, true)
+		if !ok {
+			// Block receipts not found in storage
+			return nil, false
+		}
+
+		// Find the transaction within the block
+		for idx, txn := range block.Transactions {
+			if txn.Hash == hash {
+				return toTransaction(
+					txn,
+					argUintPtr(block.Number()),
+					argHashPtr(block.Hash()),
+					&idx,
+				), true
+			}
+		}
+
+		return nil, false
 	}
-	for idx, txn := range block.Transactions {
-		if txn.Hash == hash {
-			return toTransaction(txn, block, idx), nil
+
+	// findPendingTx is a helper method for checking the TxPool
+	// for the pending transaction with the provided hash
+	findPendingTx := func() (*transaction, bool) {
+		// Check the TxPool for the transaction if it's pending
+		if pendingTx, pendingFound := e.d.store.GetPendingTx(hash); pendingFound {
+			return toPendingTransaction(pendingTx), true
 		}
+
+		// Transaction not found in the TxPool
+		return nil, false
 	}
```

Comment:
it's same. You don't need to return bool

Reply:
Excellent catch! @Kourin1996

Reply:
🤦 

Good catch 💯 

Fixed:

e92f802